pymchelper/executor/options.py

```diff
@@ -43,13 +49,23 @@ class SimulationSettings:
     (i.e. whether this is SHIELD-HIT12A input or FLUKA input)
     """
 
-    def __init__(self, input_path, simulator_exec_path=None, cmdline_opts=None):
+    def __init__(self, input_path, simulator_type = 'shieldhit', simulator_exec_path=None, cmdline_opts=None):
+        # simulator type (shieldhit, topas or fluka)
+        self.simulator_type = simulator_type
+
         # input file or directory
         self.input_path = input_path
 
-        # discover the type of MC engine based on the type of input files/directories
-        # `self._mc_environment` is set to one of the `MCEnvironment` subclasses
-        self._mc_environment = self._discover_mc_engine(input_path)
+        # set `self._mc_environment` to the proper `MCEnvironment` subclass
+        # shieldhit is default
+        if simulator_type=='shieldhit':
+            if sys.platform == 'win32':
+                self._mc_environment = SH12AEnvironmentWindows
+            self._mc_environment = SH12AEnvironmentLinux
+        elif simulator_type == 'fluka':
+            self._mc_environment = FlukaEnvironment
+        elif simulator_type == 'topas':
+            self._mc_environment = TopasEnvironment
```

Comment:
Class is publicly available, therefore:

- Seems like we have some magic strings here
- What will happen if somebody enters foobar instead `simulator_type `?
- Seems that in tests you are using enums, so maybe parameter **simulator_type** should have a SimulatorType type alias?

Additionally, in line 64 we are overriding code from line 63